### PR TITLE
fix(Skeleton): support multiple color schemes

### PR DESCRIPTION
**Pull request by brandonlilly**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [x] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [x] Tested my solution on Mobile & Tablet.
- [x] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [x] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [x] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [x] Made sure that all accessibility errors are resolved.
- [x] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [x] Added name to OWNERS file for all new components
- [x] If adding a new component, add its export to the rollup config
- [x] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**

Static keyframe names didn't play nice with multiple different color schemes existing at the same time. This switches it to use the styled-components `keyframes` api to support multiple different color schemes, and also enables skeleton variants.

### CODE REVIEW

File: src/Skeleton/index.ts [hunk 1]
```diff
@@ -1 +1 @@
-export { Skeleton } from './Skeleton.component';
+export { Skeleton, SKELETON_KEY, SKELETON_THEME } from './Skeleton.component';
```
Comment (zero0Halo) [inline src/Skeleton/index.ts:1]: I think you were just looking for an excuse to have a variable named `SKELETON_KEY`.

File: src/Skeleton/Skeleton.component.tsx [hunk 2]
```diff
@@ -1,128 +1,130 @@
...
+    colorEnd: string;
 }
 
+export const SKELETON_KEY = 'skeleton';
+export const SKELETON_THEME = {
+    variants: {
+        base: {
+            colorStart: '#E7E7E7',
+            colorEnd: '#D3D3D3',
+        },
```
Comment (zero0Halo) [inline src/Skeleton/Skeleton.component.tsx:49]: It feels a little weird  hardcoding colors in the component rather than grabbing from the theme, but I get that this is the pattern established by `Button`. Something to think about maybe.

File: src/Skeleton/Skeleton.component.tsx [hunk 3]
```diff
@@ -1,128 +1,130 @@
...
+import {
+    space as spaceStyles,
+    SpaceProps,
+    layout as layoutStyles,
+    LayoutProps,
+    borderRadius as borderRadiusStyles,
+    BorderRadiusProps,
+    variant as createVariant,
+    th,
+} from '@xstyled/system';
```
Comment (zero0Halo) [inline src/Skeleton/Skeleton.component.tsx:18]: Are the additions of `layout` and `borderRadius` to accommodate rx?

File: src/Skeleton/Skeleton.component.tsx [hunk 4]
```diff
@@ -1,128 +1,130 @@
...
+    const theme = React.useContext(ThemeContext);
+    const { colorStart: themeStart, colorEnd: themeEnd } = colorVariant({
+        ...props,
+        theme,
+    });
 
-    // This is the unicode block character that put together looks like one
-    // long bar. There are shorter and taller characters that could be used
-    // or a fatter or thinner bar. Therefore the bar thickness chould be made
-    // a prop in the future, mapping to different block characters.
```
Comment (zero0Halo) [inline src/Skeleton/Skeleton.component.tsx:124]: I wouldn't lose all of this comment. I believe `blockRatio` is the response to the second part of the comment, but the first part, `This is the unicode block character that put together looks like one long bar.`, is still helpful. Maybe move this up to where `textChar` is being assigned the character.

File: src/Skeleton/Skeleton.component.tsx [hunk 5]
```diff
@@ -134,17 +136,16 @@ export const Skeleton = ({
 
     const length = textLength || (childText && childText.length);
     const blockText =
-        length &&
-        Array(Math.ceil(length / blockToAverageCharQuotient) + 1).join(
-            blockChar
-        );
+        length && Array(Math.ceil(length * blockRatio) + 1).join(textChar);
+    const textOnly = !!blockText;
```
Comment (zero0Halo) [inline src/Skeleton/Skeleton.component.tsx:144]: I think this might be over engineering so it take it with a grain of salt (or just a point of discussion); there are a few calculations that always happen when render is triggered. Maybe worth memoizing them? I don't want to flag every spot, but there's a `replace()` with a regex, some array `join()`'s with `Math`, things of that nature. It may not make sense in some cases but I know we're all thinking more about performance so I'm throwing it out there.

**Review by zero0Halo — CHANGES_REQUESTED** [review]
Please do a follow up ticket to update the doc site. The new props and new functionality should be recorded.

File: src/Skeleton/Skeleton.component.tsx (hunk 3 above)
Comment (brandonlilly) [reply src/Skeleton/Skeleton.component.tsx:18]: Yep, figure its better to use the system

File: src/Skeleton/Skeleton.component.tsx (hunk 2 above)
Comment (brandonlilly) [reply src/Skeleton/Skeleton.component.tsx:49]: Originally it just couldn't support a theme reference and needed a raw value. This PR actually makes it so it could take `primary.light` for instance. In this case, however, its still raw because I don't think there are appropriate theme colors values to draw from (colors.background.x? doesn't really work). Better to just hardcode imo

File: src/Skeleton/Skeleton.component.tsx (hunk 5 above)
Comment (brandonlilly) [reply src/Skeleton/Skeleton.component.tsx:144]: I was going to do this but then I decided I want to early return before anything else (so that the normal case of just passing through is as inexpensive as possible). This precludes using a hook like `useMemo`

File: src/Skeleton/index.ts (hunk 1 above)
Comment (brandonlilly) [reply src/Skeleton/index.ts:1]: It wasn't intentional but I really enjoy the fact that I had to make one

**Review by zero0Halo — APPROVED** [review]
GTG
